nca/Utils/QueryOutputHandler.py

```diff
@@ -0,0 +1,395 @@
+        """
+        computes and returns the output explanation is yaml format.
+        An OutputExplanation object may have only one field representing the explanation data besides to
+        explanation_description, so the output is computed by calling the get_explanation_in_yaml method of this field
+        :return the explanation written with its description in a list of dict objects pattern which matches yaml dump
+        later.
+        A single dict pattern is {'description' : <explanation_description_str> ,
+        <key_name>: <list of examples matching the result> }
+         :rtype: list[dict]
+        """
+        for val in self.__dict__.values():
+            if not isinstance(val, str) and val is not None:
+                return val.get_explanation_in_yaml(self.explanation_description)
+
+
+# following classes describe the more possible OutputExplanation patterns, each class consists of the explanation
+# fields that may appear together in one output_explanation and additional info for writing the explanation if required
+@dataclass
+class PoliciesWithCommonPods:
+    """
+    A class for holding information of pairs of policies with common pods
+    """
+    first_policy: str = ''
+    second_policy: str = ''
+    common_pods: str = ''
+
+    def __lt__(self, other):
+        if self.first_policy == other.first_policy:
+            return self.second_policy < other.second_policy
+        return self.first_policy < other.first_policy
+
+
+@dataclass
+class IntersectPodsExplanation:
+    # used in DisjointnessQuery
+    policies_pods: list[PoliciesWithCommonPods] = None  # used in DisjointnessQuery
+
+    def get_explanation_in_yaml(self, explanation_description):
+        """
+        returns yaml format of self type of OutputExplanation
+        :param explanation_description: the relevant description of this output explanation
+        :rtype: list[dict]
+        """
+        examples = []
+        for record in self.policies_pods:
+            examples.append({'policies': [record.first_policy.split(' ')[1], record.second_policy.split(' ')[1]],
+                             'pods': record.common_pods.split(', ')})
+        return [{'description': explanation_description, 'examples': examples}]
+
+    def get_explanation_in_txt(self, explanation_description):
+        """
+        returns txt format of self type of OutputExplanation
+        :param explanation_description: the relevant description of this output explanation
+        :rtype: str
+        """
+        result = []
+        delimiter = ' '
+        for record in self.policies_pods:
+            result.append(f'{record.first_policy.split(delimiter)[0]}_1: {record.first_policy.split(delimiter)[1]}, '
+                          f'{record.second_policy.split(delimiter)[0]}_2: {record.second_policy.split(delimiter)[1]},'
+                          f' pods: {record.common_pods}')
+        return explanation_description + ':\n' + '\n'.join(result)
```

Comment:
can we simplify this code? 
a few things to consider:
- a function that gets an input policy title and returns its components (policy name, policy kind,..)
- hold common pods as PeerSet or list of peer names, to avoid splitting by some character that could be changed in the future.
- missing documentation of the expected format for policy_str (if you are relying on this in the code it should be documented).